.github/pull_request_template.md

```diff
@@ -1,10 +1,19 @@
-**Related Issue(s)**:  ...
+### Related Issues
+- fixes #issue-number
 
-**Proposed changes**:
-- ...
+### Proposed Changes:
+ <!--- In case of a bug: Describe what caused the issue and how you solved it -->
+ <!--- In case of a feature: Describe what did you add and how it works -->
 
-## Pre-flight checklist
-- [ ]  I have read the [contributors guidelines](https://github.com/deepset-ai/haystack/blob/master/CONTRIBUTING.md)
-- [ ] I have [enabled actions on my fork](https://github.com/deepset-ai/haystack/blob/master/CONTRIBUTING.md#forks)
-- [ ] If this is a code change, I added tests or updated existing ones 
-- [ ] If this is a code change, I updated the docstrings
+### How did you test it?
+<!-- unit tests, integration tests, manual verification, instructions for manual tests -->
+
+### Notes for the reviewer
+<!-- E.g. point out section where the reviewer  -->
+
+### Checklist
+- [ ] I have updated the referenced issue with new insights and changes
+- [ ] If this is a code change, I have added tests
+- [ ] I've used the [conventional commit convention](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title
```

Comment:
There's no mention of this in the contributing guidelines, let's add a paragraph there before merging